apps/web/pages/api/auth/signup.ts

```diff
@@ -1,218 +1,49 @@
-import type { NextApiRequest, NextApiResponse } from "next";
+import type { NextApiResponse } from "next";
 
-import dayjs from "@calcom/dayjs";
-import { checkPremiumUsername } from "@calcom/ee/common/lib/checkPremiumUsername";
-import { hashPassword } from "@calcom/features/auth/lib/hashPassword";
-import { sendEmailVerification } from "@calcom/features/auth/lib/verifyEmail";
+import calcomSignupHandler from "@calcom/feature-auth/signup/handlers/calcomHandler";
+import selfhostedSignupHandler from "@calcom/feature-auth/signup/handlers/selfHostedHandler";
+import { type RequestWithUsernameStatus } from "@calcom/features/auth/signup/username";
 import { IS_CALCOM } from "@calcom/lib/constants";
-import slugify from "@calcom/lib/slugify";
-import { closeComUpsertTeamUser } from "@calcom/lib/sync/SyncServiceManager";
-import { validateUsernameInTeam, validateUsername } from "@calcom/lib/validateUsername";
-import prisma from "@calcom/prisma";
-import { IdentityProvider } from "@calcom/prisma/enums";
-import { MembershipRole } from "@calcom/prisma/enums";
-import { signupSchema } from "@calcom/prisma/zod-utils";
-import { teamMetadataSchema } from "@calcom/prisma/zod-utils";
+import { HttpError } from "@calcom/lib/http-error";
 
-export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+function ensureReqIsPost(req: RequestWithUsernameStatus) {
   if (req.method !== "POST") {
-    return res.status(405).end();
+    throw new HttpError({
+      statusCode: 405,
+      message: "Method not allowed",
+    });
   }
+}
 
+function ensureSignupIsEnabled() {
   if (process.env.NEXT_PUBLIC_DISABLE_SIGNUP === "true") {
-    res.status(403).json({ message: "Signup is disabled" });
```

Comment:
What is the reason here for code moving/deleting?

Reply:
<img width="620" alt="CleanShot 2023-09-22 at 08 49 05@2x" src="https://github.com/calcom/cal.com/assets/55134778/5c155918-7d1c-47bc-b35a-4886f96e617c">
Throwing and error to catch internally - using it as a consitant way of handling errors across this route. Passing res around isnt ideal